src/atl/formula/parser.rs

```diff
@@ -0,0 +1,827 @@
+use std::str::{self, FromStr};
+use std::sync::Arc;
+
+use pom::parser::{end, Parser};
+use pom::parser::{list, one_of, seq, sym};
+
+use super::Phi;
+
+/// Parse an ATL formula
+pub(crate) fn parse_phi<'a, 'b: 'a, A: ATLExpressionParser>(
+    expr_parser: &'b A,
+    input: &'a str,
+) -> Result<Phi, String> {
+    let formula = ws() * phi(expr_parser) - end();
```

Comment:
We should allow whitespace at the end of a formula like we do at the beginning
```suggestion
    let formula = ws() * phi(expr_parser) - ws() - end();
```

Reply:
Good spot. Fixed in 0d67118